docs/modules.md

```diff
@@ -282,6 +282,47 @@ the request is a server admin.
 Modules can modify the `request_content` (by e.g. adding events to its `initial_state`),
 or deny the room's creation by raising a `module_api.errors.SynapseError`.
 
+#### Presence router callbacks
+
+Presence router callbacks allow module developers to specify additional users (local or remote)
+to receive certain presence updates from local users. Presence router callbacks can be 
+registered using the module API's `register_presence_router_callbacks` method.
+
+The available presence router callbacks are:
+
+```python 
+async def get_users_for_states(
+    self,
+    state_updates: Iterable[synapse.api.UserPresenceState],
+) -> Dict[str, Set[synapse.api.UserPresenceState]]:
+```
+**Requires** `get_interested_users` to also be registered
+
+Called when processing updates to the presence state of one or more users. This callback can
+be used to instruct the server to forward that presence state to specific users. The module
+must return a dictionary that maps from Matrix user IDs (which can be local or remote) to the
+UserPresenceState changes that they should be forwarded.
+
+Synapse will then attempt to send the specified presence updates to each user when possible.
+```python
+async def get_interested_users(self, user_id: str) -> Union[Set[str], str]
+```
+**Requires** `get_users_for_states` to also be registered
+
+Called when determining which users someone should be able to see the presence state of. This
+callback should return complementary results to `get_users_for_state` or the presence info 
+may not be properly forwarded.
+
+The callback is given a single Matrix user ID and should return a set of additional Matrix user 
```

Comment:
It would help linking this paragraph with the one above. e.g. "The callback is given the Matrix user ID of the user that's requesting presence data and is expected to return the Matrix IDs of the users they're allowed to see the presence state of". Otherwise I feel like it might be difficult for module developers to understand what the expected input and output mean from a semantics pov.